### ISSUE TITLE
Help into implementing multiple cd read speed feature from NeoRaine into fbalpha core for Retroarch

### ISSUE THREAD

**Issue by thorias** [zelurker/raine#23]
Hello,

If you can help/assist to have this nice feature of NeoRaine (having the choice to select the cd rom drive speed to a higher speed than the original neo cd system) to be implemented into the fbalpha core for Retroarch, it'll be really nice from you!

 Please see here: https://github.com/libretro/fbalpha/issues/4

It'll be also a wider recognition of the work already put in NeoRaine as the fbalpha core run in multiple devices/arch. 

Merci beaucoup pour votre aide! :-)

**Comment by zelurker** [comment]
Ah, sorry missed this mail (opened it in the phone then forgot it and it
stayed here for about 2 weeks !).

fbalpha is mainly based on mame code, right ?
So it's very different from raine code, and by the way neoraine was merged
into raine years ago...
Well when I add this feature I hadn't understood the cdrom was using a
hardware irq to work, 1st time I ever saw such hardware actually.
So in raine I use some kind of hack to emulate the cdrom, just noticed what
triggers cdrom reads in ram, and then tried to emulate it.
While loading animations are enabled, there is a counter somewhere used by
the progress bar in the animation which tells it how much it can read.
So this value is just adjusted based on the selected speed.

Now it's very different in mame, so different that I actually decided to
keep my hack about how the cdrom is working rather than using the real way
they found, because it made things actually more complex for me !
So you are on your own here since you want the best of the 2 ways of doing
it. It's probably do-able, but not obvious for sure.

Good luck and sorry for the delay !


2018-06-14 2:11 GMT+02:00 thorias <notifications@github.com>:
